### DISCUSSION TITLE
Hiding fields for Many to Many relations

### DISCUSSION THREAD

**Discussion by ookil** [MichalLytek/typegraphql-prisma#298] [Q&A]
Hello, I wanted to ask about hiding fields for Many to Many relations. When I have something like this: 

```
model Crop {
  id                 String            @id
  name               String
  buyAt              CropAtShop[]
}

model CropAtShop {
  id              String           @id
  crop            Crop             @relation(fields: [cropId], references: [id])
  shop            Shop             @relation(fields: [shopId], references: [id])
  price           Int
  /// @TypeGraphQL.omit(output: true)
  cropId          String
  /// @TypeGraphQL.omit(output: true)
  shopId          String
}

model Shop {
  id                   String           @id
  name                 String
  purchasableCrops     CropAtShop[]
}
```

I would like to omit `crop` field in `CropAtShop` when making query for crop, and similarly with shop. Basically to prevent loops. 
What I came up with is to omit `crop` in `CropAtShop` model and use that model for `Shop` but write a custom model (without crop field) and field resolver for `buyAt`.
Is this the only solution? Or could it be done differently?

**Comment by MichalLytek** [comment]
If you don't want to expose some field, just omit it from the output. Your example only shows omit on ids, have you tried on relations? Does it work with the generated field resolvers?

**Reply by ookil** [reply]
Sure, it works for relations but when I do for example 
```
model CropAtShop {
  id              String           @id
  /// @TypeGraphQL.omit(output: true)
  crop            Crop             @relation(fields: [cropId], references: [id])
  shop            Shop             @relation(fields: [shopId], references: [id])
  price           Int
  /// @TypeGraphQL.omit(output: true)
  cropId          String
  /// @TypeGraphQL.omit(output: true)
  shopId          String
}
```

then it will be omitted from both relations. 
But I would like to achieve this 
```
crop {
    id
    name
    buyAt {
      shop {
        name 
      }
      price
** crop is omitted here **
    }
  }

shop {
    id
    name
    purchasableCrops{
      crop {
        name
      }
      price
** shop is omitted here **
    }
  }
```

I'm perfectly okay doing it with custom resolvers but I was curious if there's maybe other solution

**Reply by MichalLytek** [reply]
I think your use case is too specific so that it's too narrow to be covered by some generic solution.
